consensus/blocknode.go

```diff
@@ -58,143 +41,103 @@ type blockNode struct {
 	// through 'TargetWindow' elements in a linked list.
-	var i BlockHeight
-	referenceNode := node
-	for i = 0; i < TargetWindow && referenceNode.parent != nil; i++ {
-		referenceNode = referenceNode.parent
+	var numBlocks BlockHeight
+	windowStart := child
+	for numBlocks = 0; numBlocks < TargetWindow && windowStart.parent != nil; numBlocks++ {
+		windowStart = windowStart.parent
 	}
 
 	// Calculate the amount to adjust the target by dividing the amount of time
 	// passed by the expected amount of time passed.
-	timePassed := node.block.Timestamp - referenceNode.block.Timestamp
-	expectedTimePassed := BlockFrequency * Timestamp(i)
+	timePassed := child.block.Timestamp - windowStart.block.Timestamp
+	expectedTimePassed := BlockFrequency * numBlocks
 	targetAdjustment := big.NewRat(int64(timePassed), int64(expectedTimePassed))
 
-	// Enforce a maximum target adjustment.
-	if targetAdjustment.Cmp(MaxAdjustmentUp) == 1 {
+	// Clamp adjustment to reasonable values.
+	if targetAdjustment.Cmp(MaxAdjustmentUp) > 0 {
 		targetAdjustment = MaxAdjustmentUp
-	} else if targetAdjustment.Cmp(MaxAdjustmentDown) == -1 {
+	} else if targetAdjustment.Cmp(MaxAdjustmentDown) < 0 {
 		targetAdjustment = MaxAdjustmentDown
 	}
 
 	// Multiply the previous target by the adjustment to get the new target.
-	parentTarget := node.parent.target
-	newRatTarget := new(big.Rat).Mul(parentTarget.Rat(), targetAdjustment)
-	node.target = RatToTarget(newRatTarget)
-}
-
-// heavierNode compares the depth of `newNode` to the depth of the current
-// block node, and returns true if `newNode` is sufficiently heavier, where
-// sufficiently is defined by the weight of the current block times
-// `SurpassThreshold`.
-func (s *State) heavierNode(newNode *blockNode) bool {
-	threshold := new(big.Rat).Mul(s.currentBlockWeight(), SurpassThreshold)
-	currentCumDiff := s.depth().Inverse()
-	requiredCumDiff := new(big.Rat).Add(currentCumDiff, threshold)
-	newNodeCumDiff := newNode.depth.Inverse()
-	return newNodeCumDiff.Cmp(requiredCumDiff) == 1
-}
+	newRatTarget := new(big.Rat).Mul(bn.target.Rat(), targetAdjustment)
+	child.target = RatToTarget(newRatTarget)
 
-// addBlockToTree takes a block and a parent node, and adds a child node to the
-// parent containing the block. No validation is done.
-func (s *State) addBlockToTree(b Block) (err error) {
-	parentNode := s.blockMap[b.ParentID]
-	newNode := &blockNode{
-		block:  b,
-		parent: parentNode,
-
-		height: parentNode.height + 1,
-		depth:  parentNode.childDepth(),
-
-		delayedSiacoinOutputs: make(map[SiacoinOutputID]SiacoinOutput),
-	}
-	newNode.setTarget()
-
-	// Add the node to the block map and update the list of its parents
-	// children.
-	s.blockMap[b.ID()] = newNode
-	parentNode.children = append(parentNode.children, newNode)
-
-	if s.heavierNode(newNode) {
-		err = s.forkBlockchain(newNode)
-		if err != nil {
-			return
-		}
-	}
+	// add child to parent
+	bn.children = append(bn.children, child)
 
-	return
+	return child
 }
```

Comment:
You should look at this diff. GH went through so much effort to preserve the newlines in the diff that the function is pretty much entirely unreadable. You'll get 2 lines of newChild and then a big red diff where all of heavierNode was removed. Then 2 more lines, and a big red diff where all of addBlockToTree was removed.

But hey, at least you found the minimum diff, preserved those newlines that were in both the old and new.